oidc/client.go

```diff
@@ -655,13 +655,15 @@ func chooseAuthMethod(cfg ProviderConfig) (string, error) {
 	return "", errors.New("no supported auth methods")
 }
 
-// SyncProviderConfig starts the provider config syncer
-func (c *Client) SyncProviderConfig(discoveryURL string) chan struct{} {
+// SyncProviderConfig starts the provider config syncer.  
+// This function returns a channel used to stop the sync 
+// process, as well as a function that can be used to block 
+// until the sync process has either started or been interrupted.
+func (c *Client) SyncProviderConfig(discoveryURL string) (chan struct{}, func()) {
```

Comment:
Something like that may be possible.  Though I kind of view any blocking within the library sub-optimal.  I think in this case it's much cleaner to not have the library make the decision to block here and instead allow the application to choose when that blocking will occur.  Particularly since the returned stop channel is (or seems) like a critical control.

In this case I don't have a particularly strong opinion, particularly since it sounds like this fix may be short lived.

Reply:
You convinced me...Although I dislike the blocking behavior being enforced at construction, I agree the code becomes much simpler.  I have updated the PR, let me know if you see any further improvements to add.  Thank you!